File: timetraveler-school/scripts/patch_index.py
import os
import glob
import sys

REDIRECT = '<script>(function(){var h=window.location.hash;if(h==="#/schools"||h===""||h==="#/"||h==="#"){window.location.replace("/#/login");}})()</script>'

print("=== dist/ HTML files ===")
files = glob.glob("dist/**/*.html", recursive=True) + glob.glob("dist/*.html")
for f in files:
    print(f)

if not files:
    print("ERROR: No HTML files found in dist/")
    sys.exit(1)

for filepath in files:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    if 'window.location.replace' in content:
        print("Already patched:", filepath)
        continue
    if '<script type="module"' in content:
        content = content.replace('<script type="module"', REDIRECT + '\n    <script type="module"', 1)
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print("Patched:", filepath)
    else:
        print("No module script in:", filepath)

File: timetraveler-school/scripts/patch-index.js
// Patch dist/public/index.html to redirect #/schools → #/login before app loads
const fs = require('fs');
const path = require('path');

const file = path.join('dist', 'public', 'index.html');
if (!fs.existsSync(file)) {
  // Try dist/client/index.html (TanStack Start client output)
  const alt = path.join('dist', 'client', 'index.html');
  if (fs.existsSync(alt)) {
    patchFile(alt);
  } else {
    // Search for any index.html in dist/
    const { execSync } = require('child_process');
    const found = execSync('find dist/ -name index.html 2>/dev/null || echo ""').toString().trim();
    if (found) {
      found.split('\n').forEach(f => patchFile(f.trim()));
    } else {
      console.log('No index.html found in dist/');
      process.exit(1);
    }
  }
} else {
  patchFile(file);
}

function patchFile(filePath) {
  let html = fs.readFileSync(filePath, 'utf8');
  const redirectScript = [
    '<script>',
    '(function(){',
    'var h=window.location.hash;',
    'if(h==="#/schools"||h===""||h==="#/"||h==="#"){',
    'window.location.replace("/#/login");',
    '}',
    '})();',
    '</script>'
  ].join('');
  
  if (html.includes('window.location.replace')) {
    console.log('Already patched:', filePath);
    return;
  }
  
  // Insert before first <script type="module"
  html = html.replace('<script type="module"', redirectScript + '\n    <script type="module"');
  fs.writeFileSync(filePath, html);
  console.log('Patched:', filePath);
  console.log('Script tags found:', (html.match(/<script/g) || []).length);
}
